fiftyone/operators/delegated.py

```diff
@@ -0,0 +1,178 @@
+import asyncio
+import logging
+import traceback
+
+from fiftyone.core.expressions import ObjectId
+from fiftyone.factory import DelegatedOpPagingParams
+from fiftyone.factory.repo_factory import RepositoryFactory
+from fiftyone.factory.repos import DelegatedOperationDocument
+from fiftyone.factory.repos.delegated_operation import (
+    DelegatedOperationRepo,
+)
+
+from fiftyone.operators.executor import (
+    prepare_operator_executor,
+    ExecutionResult,
+    ExecutionContext,
+    ExecutionRunState,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class DelegatedOperation(object):
+    """Base class for delegated operations.
+    Delegated operations are used to define custom operations that can be
+    applied to datasets and views.
+    Delegated operations are defined by subclassing this class and
+    implementing the :meth:`get_pipeline_stage` method.
+    """
+
+    def __init__(self, repo: DelegatedOperationRepo = None):
+        self._repo = (
+            repo
+            if repo is not None
+            else RepositoryFactory.delegated_operation_repo()
+        )
+
+    def queue_operation(
+        self,
+        operator: str,
+        delegation_target: str = None,
+        context: ExecutionContext = None,
+    ) -> DelegatedOperationDocument:
+        """Returns a queued :class:`fiftyone.core.odm.DelegatedOperationDocument` instance
+        for the operation.
+        Returns:
```

Comment:
Looks like copy + pasting of code removed some blank lines in your docstrings. Sphinx needs the whitespace in order to correctly render the docs (applies throughout PR)